WebDriverAgentLib/Utilities/FBXCAXClientProxy.m

```diff
@@ -73,14 +40,14 @@ - (BOOL)setAXTimeout:(NSTimeInterval)timeout error:(NSError **)error
                                      maxDepth:(nullable NSNumber *)maxDepth
                                         error:(NSError **)error
 {
-  NSMutableDictionary *parameters = [[NSMutableDictionary alloc] init];
+  NSMutableDictionary *parameters = [NSMutableDictionary dictionaryWithDictionary:self.defaultParameters];
   // Mimicking XCTest framework behavior (this attribute is added by default unless it is an excludingNonModalElements query)
+  // A regular snapshot retrieval path as part of resolving a query goes through [XCUIElementQuery _snapshotForElement:error:]
+  // before calling requestSnapshotForElement, and this is the part where the snapshotKeyHonorModalViews is added (as part of
+  // calling [XCTElementQuery snapshotParameters])
   // See https://github.com/appium/WebDriverAgent/pull/523
-  if (SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(@"13.0")) {
```

Comment:
btw, is it good timing to drop ios 12 support? Maybe we still address our supported device is 12.2 or around.

Reply:
yes, it makes sense to me after Xcode14 is out. We aim to support the current plus the previous version (e.g. Xcode 13 and 14 -> iOS 15/16)